### PR TITLE
[drogon] Update to 1.4.1

### PR DESCRIPTION
**Pull request by an-tao**

### CODE REVIEW

File: ports/drogon/CONTROL [hunk 1]
```diff
@@ -1,8 +1,5 @@
 Source: drogon
-Version: 1.3.0
+Version: 1.4.1
 Homepage: https://github.com/an-tao/drogon
 Description:Drogon: A C++14/17 based HTTP web application framework running on Linux/macOS/Unix/Windows
-Build-Depends: trantor, zlib, jsoncpp, libmariadb (!osx), libmariadb[iconv] (osx), libpq, sqlite3, brotli, libuuid (!windows)
-
-Feature: ctl
```
Comment (JonLiu1993) [inline ports/drogon/CONTROL:7]: Hi @an-tao ,Why did you delete this feature?

File: ports/drogon/portfile.cmake [hunk 2]
```diff
@@ -1,46 +1,42 @@
...
+  an-tao/drogon
+  REF
+  v1.4.1
+  SHA512
+  8611c18e65229095f5443f10c87e91593d619bc3a2d47da6d19d501a64be1aba12a813e44ccfc1c10179e5b0fa10121e8806955c0091258992501b88fa50d939
+  HEAD_REF
+  master
+  PATCHES
+  vcpkg.patch
+  resolv.patch)
```
Comment (JonLiu1993) [inline ports/drogon/portfile.cmake:14]: @an-tao ，Please restore the format here.

File: ports/drogon/CONTROL (hunk 1 above)
Comment (an-tao) [reply ports/drogon/CONTROL:7]: I think drogon_ctl is needed in almost all scenarios. 
Actually, some people complained to me that drogon_ctl is missing after installing drogon in the latest version of vcpkg.

File: ports/drogon/portfile.cmake [hunk 3]
```diff
@@ -1,46 +1,42 @@
...
-    PREFER_NINJA
-    OPTIONS
-        -DBUILD_EXAMPLES=OFF
-        ${FEATURE_OPTIONS}
-)
+vcpkg_configure_cmake(SOURCE_PATH
+                      ${SOURCE_PATH}
+                      PREFER_NINJA
+                      OPTIONS
+                      -DBUILD_EXAMPLES=OFF)
```
Comment (JonLiu1993) [inline ports/drogon/portfile.cmake:20]: @an-tao ，Please keep the format here the same as before.

File: ports/drogon/CONTROL (hunk 1 above)
Comment (JonLiu1993) [reply ports/drogon/CONTROL:7]: Ok that's great

**Comment by an-tao** [comment]
@JonLiu1993 If some dependencies update after the PR and break the build of drogon, what can I do to avoid this situation?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 If some dependencies update after the PR and break the build of drogon, what can I do to avoid this situation?

I'll fix it

**Review by JonLiu1993 — APPROVED** [review]

**Comment by an-tao** [comment]
> If a new problem arises in response to this situation, we will solve it, you know this cannot be avoided

I see, thanks for your reply.

File: ports/drogon/CONTROL (hunk 1 above)
Comment (ras0219-msft) [reply ports/drogon/CONTROL:7]: I think we should keep this as a feature, but make it a default:
```
Default-Features: ctl
```
This is useful for cases like cross-compiling to android or ios where creating an executable may not be possible.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR! LGTM once the feature is moved to default.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Hi @an-tao, thanks for the PR!

Obviously needing to directly link lower level postgresql libraries is not great, so I've fixed the underlying libpq port and reverted that part of the drogon patches here: https://github.com/ras0219-msft/vcpkg/commits/drogon. Could you please pull those additional commits into this PR?

Additionally, I noticed that drogon performs some automatic detection with a boost polyfill fallback -- this hidden optional dependency causes a bunch of issues in a package manager so I've taken the simplest route for now and ensured the build will fail if that path would be taken.

**Comment by an-tao** [comment]
> Hi @an-tao, thanks for the PR!
> 
> Obviously needing to directly link lower level postgresql libraries is not great, so I've fixed the underlying libpq port and reverted that part of the drogon patches here: https://github.com/ras0219-msft/vcpkg/commits/drogon. Could you please pull those additional commits into this PR?
> 
> Additionally, I noticed that drogon performs some automatic detection with a boost polyfill fallback -- this hidden optional dependency causes a bunch of issues in a package manager so I've taken the simplest route for now and ensured the build will fail if that path would be taken.

Thank you very much, I have added them to this PR, you solved a very annoying problem for me.

**Comment by ras0219-msft** [comment]
Thanks so much for the PR!
